temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java

```diff
@@ -139,34 +168,40 @@ public boolean isTerminated() {
 
   @Override
   public CompletableFuture<Void> shutdown(ShutdownManager shutdownManager, boolean interruptTasks) {
-    return poller.shutdown(shutdownManager, interruptTasks);
+    return stickyPoller != null
+        ? CompletableFuture.allOf(
+            poller.shutdown(shutdownManager, interruptTasks),
+            stickyPoller.shutdown(shutdownManager, interruptTasks))
+        : poller.shutdown(shutdownManager, interruptTasks);
```

Comment:
nit: A lot of side effects happening in one statement, which is also a return statement.

Would read easier I think with some normal conditionals and separate return.